global: Prune trailing whitespace.

Gadgetoid opened the pull request:
Prune trailing whitespace across the whole project, this is done automatically with:

```
grep -IUrl --color "[[:blank:]]$" --exclude-dir=.git --exclude=*.exp | xargs sed -i 's/[[:space:]]*$//'
```

Raised as a draft because this touches 113 files and may break things so I do not expect it to be merged as is, but I think we should be cognisant of these... issues. I've avoided .exp files, but much of the remaining diffs are still near inscrutable.

Though GitHub will filter out whitespace from diffs if you ask nicely- https://github.com/micropython/micropython/pull/13777/files?diff=unified&w=1

If there's any interest in this kind of tidyup and this PR isn’t suitable for merging, I would recommend individual authors using some variation of the above command or editor config to clean up files as they are otherwise modified.

Gadgetoid committed:
global: Prune trailing whitespace.

Prune trailing whitespace across the whole project*, this is done
automatically with:

grep -IUrl --color "[[:blank:]]$" --exclude-dir=.git --exclude=*.exp |\
xargs sed -i 's/[[:space:]]*$//'

* Skip third-party code in lib/ and drivers/cc3100/.
* Skip generated code in bluetooth_init_cc2564C_1.5.c
* Preserve command output whitespace in docs, eg:
  docs/esp8266/tutorial/repl.rst

Signed-off-by: Phil Howard <phil@gadgetoid.com>

dpgeorge commented:
Thanks, this is a reasonable thing to do.  I always try to remove any trailing space when I review/merge PRs.

But, I suggest to leave the third-party code alone, ie everything in `lib/` and `drivers/cc3100/`.

Gadgetoid commented:
Done. Updated the commit message to reflect that these directories are avoided intentionally.

robert-hh commented:
>  I always try to remove any trailing space when I review/merge PRs.

The CI yells when there are trailing spaces in the code.
But on the related topic: @projectgus prepares another global change PR by omitting the STATIC macro.  A change like that and as well this PR is a HUGE effort for everyone maintaining PRs and trying to keep it mergeable and recent. When rebasing a PR, GIT will have many merge conflicts. For whitespace these could of course be ignored. So please, if both PRs should be merged, do that immediately one after the other, such that we have the effort of updating only once.

Gadgetoid commented:
> A change like that and as well this PR is a HUGE effort for everyone maintaining PRs and trying to keep it mergeable and recent.

That's why this is still a draft, rather than something I would push to be merged. It might be less troublesome to let this cleanup happen naturally, but still useful to surface where trailing whitespace occurs. (I arrived at this PR entirely out of curiosity after experimenting with linting some other codebases.)

> The CI yells when there are trailing spaces in the code.

I guess only if it's new code or on a line that has otherwise been changed?

Gadgetoid commented:
As per my sleuthing on #13763, here are a list of open PRs which I believe *add* trailing whitespace:

| User | PR | Title |
|------|----|-------|
| NoosaHydro | https://github.com/micropython/micropython/pull/13455 | tools/pyboard.py: Fix RTS esp32 lockups. |
| nxp-yilin | https://github.com/micropython/micropython/pull/13429 | ports: mcx: Initial port for MCXN947 series MCU. |
| cwalther | https://github.com/micropython/micropython/pull/13367 | nrf/modules/machine: Implement deepsleep() |
| ricksorensen | https://github.com/micropython/micropython/pull/13276 | esp32/modesp32.c: Add mcu_temperature() function for C3/S2/S3 devices. |
| gitcnd | https://github.com/micropython/micropython/pull/13253 |  ports/esp32/boards: Add support for popular ESP32_CAM boards. |
| andrewleech | https://github.com/micropython/micropython/pull/13011 | examples/usercmodule: Add finaliser to cexample. |
| iabdalkader | https://github.com/micropython/micropython/pull/12961 | OpenAMP support. |
| robert-hh | https://github.com/micropython/micropython/pull/12937 | mimxrt: Provide WiFi and BLE support for MIMXRT boards. |
| agatti | https://github.com/micropython/micropython/pull/12853 | ports: Add RISC-V 32 bit QEMU port. |
| robert-hh | https://github.com/micropython/micropython/pull/12347 | mimxrt: Add Quadrature Encoder and Pulse Counter classes. |
| mhidro | https://github.com/micropython/micropython/pull/12281 | ports/rp2/boards: Added support for YD-RP2040 by VCC-GND studio 4 8 16 MB flash which is a pin compatible PICO clone. |
| Josverl | https://github.com/micropython/micropython/pull/12149 | CODECONVENTIONS.md: Improve docs for pre-commit setup on Windows. |
| dmazzella | https://github.com/micropython/micropython/pull/11679 | py/objint.c: Add support for int.bit_length(). |
| pschindler | https://github.com/micropython/micropython/pull/11398 | NUCLEO-H7A3ZI-Q |
| jaenrig-ifx | https://github.com/micropython/micropython/pull/11365 | New port for Infineon PSoC6 microcontrollers |
| JackAtKitronik | https://github.com/micropython/micropython/pull/11301 | port/rp2: Added WebUSB to RP2040. |
| keredson | https://github.com/micropython/micropython/pull/11183 | add set_stream to DecompIO |
| RetiredWizard | https://github.com/micropython/micropython/pull/10400 | ports/nrf: Add RTC support to NRF ports. |
| wolfpaulus | https://github.com/micropython/micropython/pull/10324 | ports/esp32/boards/PYBOX: Add support SparkFun Thing Plus board. |
| RetiredWizard | https://github.com/micropython/micropython/pull/10314 | rp2/boards: Add Waveshare rp2040 plus |
| robert-hh | https://github.com/micropython/micropython/pull/9624 | samd/adc_dac: Implememt adc.read_timed() and dac.write_timed(). |
| H-Grobben | https://github.com/micropython/micropython/pull/9367 | SMT32G4: Add USB, QPSI and AEMICS Board PYglet |
| Molorius | https://github.com/micropython/micropython/pull/9356 | shared/tinyusb: (just rp2 for now) change USB device settings and add HID |
| andrewleech | https://github.com/micropython/micropython/pull/9309 | stm32/btstack: Add support for btstack on WB55 / rfcore. |
| lowfatcode | https://github.com/micropython/micropython/pull/9221 | docs/library/time.rst: fix description of return results in time.mktime() |
| mdaeron | https://github.com/micropython/micropython/pull/9140 | tools/mpremote: Fix auto-connection bug in macOS and add config option to filter ports eligible for auto-connection. |
| r4d10n | https://github.com/micropython/micropython/pull/8637 | ports/wch: Add support for WCH CH32V307. |
| mattytrentini | https://github.com/micropython/micropython/pull/8493 | esp32/boards: Add INKPLATE6 board. |
| jbbjarnason | https://github.com/micropython/micropython/pull/8480 | pep 570 positional only parameters. |
| vikmal-sw | https://github.com/micropython/micropython/pull/8435 | tools/pyboard.py: ProcessToSerial cross-platform fix |
| andrewleech | https://github.com/micropython/micropython/pull/8318 | nrf/bluetooth: Add support for nimble based ubluetooth. |
| harbaum | https://github.com/micropython/micropython/pull/8253 | Proposal: Esp32 machine.PCNT through Encoder.py and Counter.py |
| Romaric-RILLET | https://github.com/micropython/micropython/pull/8236 | mimxrt: Allow to choose the pin to use for miso and cs |
| andrewleech | https://github.com/micropython/micropython/pull/8231 | tools/mpremote: Add manifest function. |
| alphaFred | https://github.com/micropython/micropython/pull/8229 | mimxrt/mboot: Adds bootloader support. |
| andrewleech | https://github.com/micropython/micropython/pull/8152 | extmod/re: Use buffer protocol for data to search through. |
| andrewleech | https://github.com/micropython/micropython/pull/8027 | extmod/modbluetooth: Add gap_indicate_service_changed. |
| MarceauFillon | https://github.com/micropython/micropython/pull/7877 | extmod/nimble: Support database hash storing |
| lorcap | https://github.com/micropython/micropython/pull/7859 | Add new module docs/library/datetime |
| andrewleech | https://github.com/micropython/micropython/pull/7845 | extmod/modbluetooth: Add gap_unpair command. |
| MrJake222 | https://github.com/micropython/micropython/pull/7784 | ports/esp8266: Software Serial (SoftUART) support. |
| caco3 | https://github.com/micropython/micropython/pull/7725 | Add support to read/write a single block in the RTC Memory |
| marcidy | https://github.com/micropython/micropython/pull/7526 | Introduce Non-blocking wifi scan for ESP32 |
| karfas | https://github.com/micropython/micropython/pull/7133 | esp32/machine_rtc: ESP32 machine.RTC().usermem() returning a bytearray. |
| ekondayan | https://github.com/micropython/micropython/pull/7048 | esp32/ota: Implement ESP-IDF OTA functionality. |
| steven-michalske | https://github.com/micropython/micropython/pull/6963 | stm32/can: Add support for a software based CAN message buffer. |
| sanyi | https://github.com/micropython/micropython/pull/6894 | Add RPI PIO example for quadrature encoder. |
| IhorNehrutsa | https://github.com/micropython/micropython/pull/6639 | ESP32: New hardware PCNT Counter() and Encoder(). |
| nickovs | https://github.com/micropython/micropython/pull/6389 | Support for AES GCM mode |
| andrewleech | https://github.com/micropython/micropython/pull/6080 | ports/unix: Add full uos.dupterm support. |
| stinos | https://github.com/micropython/micropython/pull/6055 | windows: Publish executables from Appveyor builds available (v2) |
| tschmid | https://github.com/micropython/micropython/pull/5514 | esp32/DAC: Add cosine generator capability |
| glennrub | https://github.com/micropython/micropython/pull/5104 | nrf/nfc: Add basic support for NFC tag 2. |
| livius2 | https://github.com/micropython/micropython/pull/3583 | Simple font size scaling for framebuf |
| martinribelotta | https://github.com/micropython/micropython/pull/3400 | lpc: Add basic micropython port to LPC Cortex-M series microcontrollers |

robert-hh commented:
I do not recall at which time the check for trailing whitespace in code files was added to the CI, but since then it should only slip though in documentation files. Maybe in makefiles as well.

projectgus commented:
Thanks @Gadgetoid! Agree it's best if we can merge this PR and #13763 together to minimise churn.

dpgeorge commented on ports/stm32/boards/LEGO_HUB_NO6/bluetooth_init_cc2564C_1.5.c:
```diff
@@ -44,530 +44,530 @@ const uint8_t cc256x_init_script[] = {
     // #################################################################
```
This file is generated code.  Please don't change it.

dpgeorge commented on docs/develop/gettingstarted.rst:
```diff
@@ -112,10 +112,10 @@ Check that you have Python available on your system:
+   Python 3.5.0 (default, Jul 17 2020, 14:04:10)
    [GCC 5.4.0 20160609] on linux
    Type "help", "copyright", "credits" or "license" for more information.
-   >>> 
```
This is technically correct to have a trailing space, because that's what is output at the REPL.  So I suggest leaving it there.

dpgeorge commented on docs/esp8266/tutorial/repl.rst:
```diff
@@ -144,9 +144,9 @@ function::
     ...    p.value(not p.value())
-    ...    
-    ...    
-    ...    
```
As above, it's correct to have spaces here because that's what is output at the prompt.

dpgeorge commented on docs/esp8266/tutorial/repl.rst:
```diff
@@ -169,9 +169,9 @@ print some text instead of calling toggle, to see the effect)::
     ...     time.sleep_ms(500)
-    ...    
-    ...    
-    ...    
```
These spaces should remain.

dpgeorge commented on docs/esp8266/tutorial/repl.rst:
```diff
@@ -189,7 +189,7 @@ paste a chunk of text into the REPL.  If you press ctrl-E you will see the
 
     paste mode; Ctrl-C to cancel, Ctrl-D to finish
-    === 
+    ===
```
This space should remain.

dpgeorge commented on ports/cc3200/simplelink/oslib/osi.h:
```diff
@@ -3,35 +3,35 @@
 // MACRO and Function prototypes for TI-RTOS and Free-RTOS API calls
 //
-// Copyright (C) 2014 Texas Instruments Incorporated - http://www.ti.com/ 
-// 
```
This simplelink is third-party code.  But I guess if we are going to add CI to check for trailing spaces, then probably it's easier to just remove these as well.

robert-hh replied:
> if we are going to add CI to check for trailing 

Isn't that already done. I have seen some complaints of the code formatting checker about trailing space.

Gadgetoid replied:
If it's third party code, then `ports/cc3200/simplelink/oslib/osi.h` is unlikely to be changed outside of total replacement? Thus it wont trigger the whitespace checks. Is it best left untouched?

afaik the only whitespace enforcement is via uncrustify, and just running that across the whole codebase results in yet another *huge* list of small code consistency changes- https://github.com/micropython/micropython/commit/18d0c7f0edcd7d7eaedfae0e762e8959af00fade

dpgeorge replied:
> and just running that across the whole codebase results in yet another _huge_ list of small code consistency changes

You must be running it wrong, because the macro indent/dedent changes should not be there.  You need to run `tools/codeformat.py`.

Gadgetoid replied:
I was running:

```
find . -name *.c -not -path "./lib/*" -not -path "./ports/cc3200/FreeRTOS/*" -not -path "./drivers/cc3100/*" | xargs uncrustify -c tools/uncrustify.cfg -lC --replace --no-backup
```
But on closer inspection it looks like `tools/codeformat.py` has code (`fixup_c() `) to then *undo* these erroneous changes.

I tried running codeformat *after* directly running uncrustify and received a much smaller set of changes:

https://github.com/pimoroni/micropython/commit/052e0ad1a3a52923d582f95df9b63d5247811d15

Weirdly if I just run something like:

```
find . -name *.c -not -path "./lib/*" -not -path "./ports/cc3200/FreeRTOS/*" -not -path "./drivers/cc3100/*" | ./tools/codeformat.py -v -c -f
```

I get a long list of files that have ostensibly been considered and changed, but no actual changes... it was always my assumption that `codeformat.py` only picked up files changed in the last commit, but I can't actually find a mechanism for that.

Poking a little further, it looks like all of my changes are to files that are *NOT* in `PATHS` and not in `EXCLUSIONS˘, eg:  `drivers/bus/softspi.c` or `examples/natmod/btree/btree_c.c`.

I should probably raise another PR to consider these...

Gadgetoid replied:
PR raised here, since there are a few minor but salient issues that shouldn't be lost in this nested chat - https://github.com/micropython/micropython/pull/14046

Gadgetoid commented:
Rebased, and added details of the additional exceptions to the commit message for posterity.

dpgeorge commented:
Merged in dda9b9c6da5d3c31fa8769e581a753e95a270803
